From 1cd208fc4d33532fad469dc889a2beed5be95908 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Wed, 21 Mar 2012 15:34:20 -0700
Subject: [PATCH] libparted: use largest_partnum in dm_reread_part_table
 (#803108)

DM devices can have more than 16 partitions, notify the kernel about
all of them.

* libparted/arch/linux.c (dm_reread_part_table): Use largest_partnum

Resolves: rhbz#803108
---
 libparted/arch/linux.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libparted/arch/linux.c b/libparted/arch/linux.c
index 7dd664e..ed14f50 100644
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -2516,14 +2516,13 @@ _dm_reread_part_table (PedDisk* disk)
           return 1;
 
         int     rc = 1;
-        int     last = PED_MIN (largest_partnum, 16);
         int     i;
 
         sync();
         if (!_dm_remove_parts(disk->dev))
                 rc = 0;
 
-        for (i = 1; i <= last; i++) {
+        for (i = 1; i <= largest_partnum; i++) {
                 PedPartition*      part;
 
                 part = ped_disk_get_partition (disk, i);
-- 
1.7.11.4

